src/storage.ts

```diff
@@ -0,0 +1,684 @@
+import * as annotator from 'annotator';
+import {IAppInstance} from "annotator";
+
+const {util} = annotator;
+const {$, Promise} = util;
+
+
+// id returns an identifier unique within this session
+const id = (function () {
+    let counter = -1;
+    return function () {
+        return counter += 1;
+    };
+}());
+
+/**
+ * function:: debug()
+ *
+ * A storage component that can be used to print details of the annotation
+ * persistence processes to the console when developing other parts of
+ * Annotator.
+ *
+ * Use as an extension module::
+ *
+ *     app.include(annotator.storage.debug);
+ *
+ */
+export const debug = function () {
+    function trace(action:string, annotation: any) {
+        const copyAnno = JSON.parse(JSON.stringify(annotation));
+        console.debug("annotator.storage.debug: " + action, copyAnno);
+    }
+
+    return {
+        create: function (annotation: any) {
+            annotation.id = id();
+            trace('create', annotation);
+            return annotation;
+        },
+
+        update: function (annotation: any) {
+            trace('update', annotation);
+            return annotation;
+        },
+
+        'delete': function (annotation: any) {
+            trace('destroy', annotation);
+            return annotation;
+        },
+
+        query: function (queryObj: any) {
+            trace('query', queryObj);
+            return {results: [], meta: {total: 0}};
+        },
+
+        configure: function (registry: annotator.registry.Registry) {
+            registry.registerUtility(this, 'storage');
+        }
+    };
+};
+
+
+/**
+ * function:: noop()
+ *
+ * A no-op storage component. It swallows all calls and does the bare minimum
+ * needed. Needless to say, it does not provide any real persistence.
+ *
+ * Use as a extension module::
+ *
+ *     app.include(annotator.storage.noop);
+ *
+ */
+export const noop = function () {
```

Comment:
to można by chyba w formie klasy zapisać?

Reply:
Sprawdziłem, chyba nie za bardzo. To przez `annotator.Apps` gdzie nie jest używane `new`
```js
App.prototype.include = function (module, options) {
    var mod = module(options);
    if (typeof mod.configure === 'function') {
        mod.configure(this.registry);
    }
    this.modules.push(mod);
    return this;
};
```